src/foam/dao/index/TreeNode.java

```diff
@@ -383,6 +383,40 @@ protected void select_(TreeNode currentNode, Sink sink, long skip, long limit, l
     }
   }
 
+  protected long[] groupBy(TreeNode currentNode, Sink sink, long skip, long limit, long size, Index tail) {
```

Comment:
I don't think skip and limit work well with GroupBy, or getting it to work properly would be extremely complicated. Which is fine, because I've never seen them used together anyway. I would make the GroupByPlan only be used if skip and limit aren't set. Otherwise, use a SkanPlan.